moveit_ros/planning/planning_scene_monitor/include/moveit/planning_scene_monitor/planning_scene_monitor.h

```diff
@@ -337,6 +338,13 @@ class PlanningSceneMonitor : private boost::noncopyable
    */
   bool requestPlanningSceneState(const std::string& service_name = DEFAULT_PLANNING_SCENE_SERVICE);
 
+  /** @brief Create an optional service for getting the complete planning scene
+   *         This is useful for satisfying the Rviz PlanningScene display's need for a service
+   *         without having to use MoveGroup
+   *  @param service_name The topic to provide the service
+   */
+  void providePlanningSceneService(const std::string& service_name = DEFAULT_PLANNING_SCENE_SERVICE);
```

Comment:
Wouldn't the default mean that we start a service that the service client in the same PSM instance would subscribe to?

Reply:
I've just added a new commit with lots of explanations and warnings about this scenario.

Overall, yes, PSM is now capable of being a client or a server, but you shouldn't be both.